### ISSUE TITLE
Disabled menu items should be focusable

### ISSUE THREAD

**Issue by gnomeontherun** [vmware/clarity#3602]
## Describe the bug

Disabled menu items should be focusable. Since the Dropdown component is implemented as an ARIA menu, the implementation should follow the recommendations at https://www.w3.org/TR/wai-aria-practices-1.1/#menu. wai-aria-practices-1.1 recommends to ensure that disabled menu items are focusable but cannot be activated. Disabled menu items should also be implemented with aria-disabled. 

## How to reproduce

https://stackblitz.com/github/clane/clarityTest2/tree/master/clarity-test-app

- Locate the "Dropdown Examples" heading
- Using the keyboard alone, open the dropdown under the 'Positioning Demo' heading
- Using the arrow keys, try to move focus/active-descendant to the disabled menu item and observe that you cannot

## Expected behavior

Ensure that disabled menu items are focusable(or referenced as the active-descendant) but cannot be activated as described at https://www.w3.org/TR/wai-aria-practices-1.1/#menu.

Implement all disabled menu items aria-disabled="true". The value can be either be set to false when the menu item is no longer disabled, or the aria-disabled attribute can be removed. See https://www.w3.org/TR/wai-aria-1.1/#aria-disabled.

## Versions

**App**

* Angular: [e.g. 6]
* Node: [e.g. 8.10.0]
* Clarity: [e.g. 0.12.5]

**Device:**

* Type: [e.g. MacBook]
* OS: [e.g. iOS]
* Browser [e.g. chrome, safari]
* Version [e.g. 22]

## Additional notes

_Add any other notes about the problem here._

**Comment by jeeyun** [comment]
As per https://www.w3.org/TR/wai-aria-practices-1.1/#kbd_disabled_controls, I will make the changes to make a disabled menu item focusable.

@colinreedmiller, would you be able to provide UX guidance on "focused" color for disabled item? Currently we don't have this and I think we'd need to give sighted user a visual indication when a disabled item has focus.

**Comment by Jinnie** [comment]
Not sure if we shouldn't go back to standard outline focus indicator. It will play well with selected/hover states. It would be nice if @chlane gave some comment about this. I think we had some informal talk about it recently. We'd better have common user experience in similar components, like the tabs overflow, dg-actions, etc.

**Comment by jeeyun** [comment]
After a lot of discussion with design and a11y, we are going to use the standard browser focus indicator for dropdown menu items.
